android/src/main/java/com/rnmapbox/rnmbx/components/mapview/RNMBXMapView.kt

```diff
@@ -1097,6 +1095,50 @@ open class RNMBXMapView(private val mContext: Context, var mManager: RNMBXMapVie
         }
     }
 
+    fun setFeatureState(
+      featureId: String,
+      state: HashMap<String, Value>,
+      sourceId: String,
+      sourceLayerId: String?,
+      response: CommandResponse
+    ) {
+        mapView.getMapboxMap().setFeatureState(sourceId, sourceLayerId, featureId, Value.valueOf(state))
+        response.success { }
+    }
+
+    fun getFeatureState(
+      featureId: String,
+      sourceId: String,
+      sourceLayerId: String?,
+      response: CommandResponse
+    ) {
+        mapView.getMapboxMap().getFeatureState(sourceId, sourceLayerId, featureId) { expected ->
+            if (expected.isValue) {
+                response.success {
+                    val state = expected.value!!.contents
```

Comment:
Want to avoid using `!!` when possible, since it will throw a NullPointerException. You should be good using:
```suggestion
                    val state = expected.value?.contents
```

and allowing any instances where `state == null` to fallback to the else-block